handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -306,6 +306,16 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
+     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
+     * users can override this method to implement custom behavior.
+     *
+     * @see AsyncMapping#map(Object, Promise)
+     */
+    protected Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
```

Comment:
@normanmaurer @Scottmitch please let me know if you want to make changes to it.

On my end I don't need the mapper to be exposed as I'd pull it via ctor if necessary but the above code example is the most likely path I'll go in terms of using it.

Also, please take a look at #5787 if you haven't already.